indico/modules/events/management/controllers.py

```diff
@@ -265,8 +265,10 @@ class RHManageEventPersonLinks(RHConferenceModifBase):
 
     def _process(self):
         form = EventPersonLinkForm(obj=self.event_new, event=self.event_new, event_type=self.event_new.type)
+        non_users_before = find_unregistered_users(self.event_new.person_links)
         if form.validate_on_submit():
             update_event(self.event_new, form.data)
+            flash_if_unregistered(non_users_before, self.event_new.person_links, self.event_new)
```

Comment:
I would change this function to act as a [context manager](https://docs.python.org/2/library/contextlib.html#contextlib.contextmanager):

```python
with flash_if_unregistered(self.event_new, self.event_new.person_links):
    update_event(self.event_new, form.data)
```

Implementation-wise, you'd do something like this:

```python
@contextmanager
def flash_if_unregistered(...):
    old_non_users = ...
    yield
    new_non_users = ...
    # now do what you currently do (compare the two lists and flash if needed)
```